framework/math/linear_solver/linear_solver.h

```diff
@@ -78,29 +48,54 @@ class LinearSolver
   void SetKSPSolveSuppressionFlag(bool flag) { suppress_kspsolve_ = flag; }
   bool GetKSPSolveSuppressionFlag() const { return suppress_kspsolve_; }
 
+  /**
+   * Set up the linaer solver
+   */
+  virtual void Setup();
+
+  /**
+   * Solve the system
+   */
+  virtual void Solve();
+
 protected:
+  bool IsSystemSet() const { return system_set_; }
   virtual void PreSetupCallback();
   virtual void SetOptions();
   virtual void SetSolverContext();
   virtual void SetConvergenceTest();
   virtual void SetMonitor();
   virtual void SetPreconditioner();
-
   virtual void SetSystemSize() = 0;
   virtual void SetSystem() = 0;
   virtual void PostSetupCallback();
-
-public:
-  virtual void Setup();
-
-protected:
   virtual void PreSolveCallback();
   virtual void SetInitialGuess() = 0;
   virtual void SetRHS() = 0;
   virtual void PostSolveCallback();
 
-public:
-  virtual void Solve();
+  /// Solver name
+  const std::string solver_name_;
+  const std::string iterative_method_;
+
+  LinSolveContextPtr context_ptr_ = nullptr;
+
+  /// Matrix
+  Mat A_;
+  /// Right-hand side
+  Vec b_;
+  /// Solution vector
+  Vec x_;
+  /// PETSc KSP object
+  KSP ksp_;
+  /// Number of local DOFs in the system
+  int64_t num_local_dofs_ = 0;
+  /// Number of global DOFs in the system
```

Comment:
```suggestion
```
Redundant

Reply:
I'm ok with removing comments but I see /// so isn't that comment used in the doxy? Generally, we need to discuss documentation soon, before the end of the year. Meanwhile, I'd say do not change it (unless I've misunderstood something!)

Reply:
Yes, those comments go to doxy. The reason I say to remove them is solely based on the principal that obvious things do not need documentation. In this case, the doxy will look something like
```
num_local_dofs_
Number of local DOFs in the system.
```
which does not add anything of value.